components/shapediver/parameter/ParameterSelectionComponent.tsx

```diff
@@ -22,87 +22,150 @@ export default function ParameterSelectionComponent(props: PropsParameter) {
 	const sessionApi = useShapeDiverStoreViewer(state => { return state.sessions[props.sessionId]; });
 	const parameterApi = sessionApi?.getParameterByName(props.parameterId)[0];
 
-	let acceptRejectMode = props.acceptRejectMode;
-	if(parameterApi && isInteractionSelectionParameterSettings(parameterApi.settings as any)) {
-		const selectionProps = parameterApi.settings?.props as ISelectionParameterSettings;
-		if((selectionProps?.minimumSelection ?? 1) < (selectionProps?.maximumSelection ?? 1)) {
-			acceptRejectMode = true;
-		}
-	}
+	const selectionProps = parameterApi.settings?.props as ISelectionParameterSettings;
+	const minimumSelection = selectionProps?.minimumSelection ?? 1;
+	const maximumSelection = selectionProps?.maximumSelection ?? 1;
 
 	const {
+		actions,
 		definition,
 		handleChange,
 		onCancel,
 		disabled,
 		value,
 		state
-	} = useParameterComponentCommons<string>({
-		acceptRejectMode,
-		disableIfDirty: props.disableIfDirty,
-		sessionId: props.sessionId,
-		parameterId: props.parameterId,
-		overrides: props.overrides,
-	}, 0);
+	} = useParameterComponentCommons<string>(props);
 	
 	// state for the selection application
 	const [selectionActive, setSelectionActive] = useState<boolean>(false);
-	const { selectedNodes, resetSelectedNodeNames } = useSelection(props.sessionId, VIEWPORT_ID, selectionActive ? definition.settings as IInteractionParameterSettings : undefined);
+	// state for if the current value can be accepted
+	const [acceptableValue, setAcceptableValue] = useState<boolean>(false);
 
-	useEffect(() => {
-		const parameterValue: SelectionParameterValue = { names: selectedNodes };
-		handleChange(JSON.stringify(parameterValue));
-	}, [selectedNodes]);
+	const { selectedNodeNames, setSelectedNodeNames } = useSelection(props.sessionId, VIEWPORT_ID, selectionActive ? definition.settings as IInteractionParameterSettings : undefined);
 
 	useEffect(() => {
-		if(acceptRejectMode === true && state.execValue === value) {
+		// check if the current value is acceptable
+		const acceptable = selectedNodeNames.length >= minimumSelection && selectedNodeNames.length <= maximumSelection;
+		setAcceptableValue(acceptable);
+
+		// case where the confirm/cancel buttons are not needed
+		if(minimumSelection === maximumSelection && acceptable) {
 			setSelectionActive(false);
-			resetSelectedNodeNames();
+			const parameterValue: SelectionParameterValue = { names: selectedNodeNames };
+			actions.setUiValue(JSON.stringify(parameterValue));
+			actions.execute(!props.acceptRejectMode);
```

Comment:
@MajorMeerkatThe3rd I suggest to use `handleChange` from the `useParameterComponentCommons` hook using 0 for the second parameter (timeout)